tests/internal/test_glob_matcher.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from ddtrace.internal.glob_matching import GlobMatcher
+
+
+@pytest.mark.parametrize(
+    "pattern,string,result",
+    [
+        ("test_string", "test_string", True),
+        ("test_string", "a_test_string_a", False),
+        ("test_st?ing", "test_string", True),
+        ("test_st?i?g", "test_string", True),
+        ("test_str*", "test_string", True),
+        ("t?st_str*", "test_string", True),
+        ("t?st_str*", "test_string", True),
+        ("?est_string", "test_string", True),
+        ("test_strin?", "test_string", True),
+        ("test?_string", "test_string", False),  # Test empty string for ?
+        ("test_s*ring", "test_string", True),  # Test empty string for *
+        ("*", "test_string", True),
+        ("a*", "a**", True),
+        ("foo.*", "foo.you", True),
+        ("foo.*", "snafoo", False),
+        ("*stuff", "lots of stuff", True),
+        ("*stuff", "stuff to think about", False),
+        ("*a*a*a*a*a*a", "aaaaaaaaaaaaaaaaaaaaaaaaaax", False),
+        ("*a*a*a*a*a*a", "aaaaaaaarrrrrrraaaraaarararaarararaarararaaa", True),
```

Comment:
We should add parameters to ensure bracket expansion and escape characters are not supported.